test/expectations/expectations.go

```diff
@@ -0,0 +1,63 @@
+package expectations
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"strings"
+)
+
+func BuildUpExpectations(folder string) map[string][]string {
```

Comment:
So I think the testing framework for the whole generation testing should be different. At the moment it works afaik like this:
- Define C code which should be analyzed
- Define Go constants as string literals
- During test:
- Look up Go constants using the AST
- Parse the generated files
- Look up string constants in files

I think this is not good for these reasons:
- It is too complicated
- There is no gofmt, typesafety, ... for string literals, but for there is safety for Go code -> it is not easy to adapt
- We are not testing if there are other generated files/content
- We are not testing the order of the outcome which should be deterministic

So I am proposing an easier solution:
- Define C code in C files
- Define the generated Go code in Go files
- Generate into a temporary folder
- Compare all content of this folder all with all of the generated content (we could even use a diff here)

This is not only safer and easier to adapt but we also see the difference on why something went wrong which cannot be done using string.contains.